extension/src/countminsketch.rs

```diff
@@ -0,0 +1,143 @@
+                .collect();
+
+            let mut counters: Vec<Vec<i64>> = Vec::with_capacity(self.depth as usize);
+            for row in 0..self.depth {
+                let row_start = (row * self.width) as usize;
+                let row_width = self.width as usize;
+                counters.push(
+                    self.counters
+                        .iter()
+                        .skip(row_start)
+                        .take(row_width)
+                        .collect(),
+                );
+            }
+
+            CountMinSketchInternal::new(
+                self.width as usize,
+                self.depth as usize,
+                hashfuncs,
+                counters,
+            )
+        }
+
+        pub fn from_internal_countminsketch(sketch: &CountMinSketchInternal) -> Self {
+            CountMinSketch::new(
+                sketch.width().try_into().unwrap(),
+                sketch.depth().try_into().unwrap(),
+                sketch.hash_keys(),
+                sketch.counters().iter().cloned().flatten().collect(),
+            )
+        }
+    }
+
+    ron_inout_funcs!(CountMinSketch);
+}
+
+use toolkit_experimental::CountMinSketch;
+
+#[aggregate]
+impl toolkit_experimental::count_min_sketch {
+    type State = CountMinSketchInternal;
+
+    fn transition(
+        state: Option<State>,
+        #[sql_type("text")] value: Option<String>,
+        #[sql_type("float")] error: f64,
+        #[sql_type("float")] probability: f64,
+    ) -> Option<State> {
+        let value = match value {
+            None => return state,
+            Some(value) => value,
+        };
+
+        let mut state = match state {
+            None => CountMinSketchInternal::with_prob(error, probability).into(),
+            Some(state) => state,
+        };
+
+        state.add_value(value);
+        Some(state)
+    }
+
+    fn finally(state: Option<&mut State>) -> Option<CountMinSketch<'static>> {
+        Some(CountMinSketch::from_internal_countminsketch(
+            &(state.unwrap()),
+        ))
+    }
+
+    const PARALLEL_SAFE: bool = true;
+
+    fn serialize(state: &mut State) -> bytea {
+        crate::do_serialize!(state)
+    }
+
+    fn deserialize(bytes: bytea) -> State {
+        crate::do_deserialize!(bytes, State)
+    }
+
+    fn combine(state1: Option<&State>, state2: Option<&State>) -> Option<State> {
+        match (state1, state2) {
+            (None, None) => None,
+            (None, Some(only)) | (Some(only), None) => Some(only.clone()),
+            (Some(a), Some(b)) => {
+                let (mut a, b) = (a.clone(), b.clone());
+                a.combine(b);
+                Some(a)
+            }
+        }
+    }
+}
+
+#[pg_extern(immutable, parallel_safe, schema = "toolkit_experimental")]
+pub fn approx_count(item: String, aggregate: Option<CountMinSketch>) -> i64 {
+    let sketch = aggregate.unwrap();
+    CountMinSketch::to_internal_countminsketch(&sketch).estimate(item)
+}
```

Comment:
We will need some pg_tests showing that this works correctly in Postgres.  You should be able to mostly copy a `SPI::execute` type test from any of the other aggregate rs files under extension.

Reply:
tests added :)